### PR TITLE
Fix views_path macro, call macro, and add support for nested view folders

### PR DESCRIPTION
**Pull request by trapped**
Fixes ysbaddaden/artanis#3 and ysbaddaden/artanis#4.

### CODE REVIEW

File: src/render.cr [hunk 1]
```diff
@@ -10,18 +10,18 @@ module Artanis
...
       {% else %}
-        render_{{ name.id }}_{{ engine.id }}
+        render_{{ name.gsub(/\//, "__SLASH__").id }}_{{ engine.id }}
       {% end %}
     end
 
     macro views_path(path)
       {%
-        views = `cd #{ path } 2>/dev/null && ls *.ecr || echo -n ""`
+        views = `cd #{ path } 2>/dev/null && find . -name "*.ecr" -type f | cut -c 3- || true`
```
Comment (ysbaddaden) [inline src/render.cr]: Why cut?

Comment (trapped) [inline src/render.cr]: `find` prints a relative path with a leading `./`; `cut -c 3-` removes the `./` by selecting the text after (and including) the third character to the end of the line. You could do the same in Crystal with `.gsub(/^\.\//, "")`.

Comment (ysbaddaden) [inline src/render.cr]: Let's avoid the shell command and use Crystal instead. We're already abusing the `run` command too much :)

**Comment by trapped** [comment]
I've updated my fork to replace that `cut` command with a `gsub`. Anything else that you think should be changed?

**Comment by ysbaddaden** [comment]
Thanks!

**Comment by ysbaddaden** [comment]
I hit the wrong button :/

**Pull request merged**